hello-compose-app/src/main/kotlin/hello/com/pose/MainScreen.kt

```diff
@@ -15,42 +13,82 @@ import androidx.compose.ui.unit.dp
 import androidx.lifecycle.viewmodel.compose.viewModel
 import coil.ImageLoader
 import coil.compose.rememberAsyncImagePainter
+import hello.com.pose.composable.Alert
 import hello.com.pose.composable.SearchBar
 import hello.com.pose.shared.domain.photo.Photo
+import kotlinx.coroutines.launch
 
 @Composable
 fun MainScreen(viewModel: MainViewModel = viewModel()) {
-    val query: MutableState<String> = remember { mutableStateOf("") }
+    val lazyGridState = rememberLazyGridState()
+    val coroutineScope = rememberCoroutineScope()
+    val currentSearchQuery by viewModel.searchQuery.collectAsState()
+
+    val shouldStartPaginate = remember {
+        derivedStateOf {
+            viewModel.canPaginate && lazyGridState.isLastItemVisible
+        }
+    }
+
+    LaunchedEffect(key1 = shouldStartPaginate.value) {
+        if (shouldStartPaginate.value && viewModel.pagingState == PagingState.IDLE) {
+            viewModel.getPhotosByQuery(currentSearchQuery)
+        }
+    }
 
     Surface(modifier = Modifier.fillMaxSize()) {
         Column(modifier = Modifier.padding(8.dp)) {
-            SearchBar(text = query.value,
+            SearchBar(text = currentSearchQuery,
                 inputChange = {
-                    query.value = it
-                    viewModel.getImageList(query.value)
+                    viewModel.setNewQuery(it)
                 })
-
-            LazyVerticalGrid(columns = GridCells.Fixed(4)) {
-                viewModel.photos.value.data.let { photo ->
-                    items(photo) {
-                        MainContentItem(photo = it)
+            SearchList(viewModel, lazyGridState)
+            LaunchedEffect(key1 = currentSearchQuery) {
+                if (viewModel.prevQuery != currentSearchQuery) {
+                    coroutineScope.launch {
+                        lazyGridState.animateScrollToItem(index = 0)
                     }
                 }
             }
         }
     }
 }
 
+@Composable
+fun SearchList(viewModel: MainViewModel, gridState: LazyGridState) {
+    LazyVerticalGrid(
+        columns = GridCells.Fixed(4),
+        state = gridState
+    ) {
+        items(viewModel.photoList) { photo ->
+            MainContentItem(photo = photo)
```

Comment:
컴포저블 이름과 하나의 인자로 값으로 명확하게 표시되는 코드는 인자를 명시하지 않아도 될 것 같습니다!

Reply:
그렇군요 무조건 인자로 명시해주는게 좋다고 생각했습니다 수정하겠습니다!
```suggestion
            MainContentItem(photo = it)
```

Reply:
제가 전달을 잘못했네요 😅  MainContentItem(photo)를 의미한거였습니다!